beep/structure/cli.py

```diff
@@ -137,7 +139,7 @@ def process_file_list_from_json(file_list_json, processed_dir="data-share/struct
             new_filename = add_suffix_to_filename(new_filename, "_structure")
             structured_run_loc = os.path.join(processed_dir, new_filename)
             structured_run_loc = os.path.abspath(structured_run_loc)
-            dumpfn(dp, structured_run_loc)
+            dp.to_json_file(structured_run_loc, as_legacy=omit_raw)
```

Comment:
@ardunn what is the reason for using different args `as_legacy` and `omit_raw` for the same purpose?

Reply:
If I want to troubleshoot some issues with the structuring and need the `raw_data` attribute again after it has been dropped, would I just have to use `from_file` method to repopulate it [since the paths to raw_data will still be stored?]?

Reply:
## Differences in naming

> @ardunn what is the reason for using different args `as_legacy` and `omit_raw` for the same purpose?

Good observation, I will change them to be the same. Would it be ok with you to just call everything `as_legacy`? Or alternatively, `omit_raw`? If `omit_raw` I'd want to change the BEEPDatapath syntax to all be `omit_raw` instead of `legacy`. 

On second thought, that might make things really complicated, since the legacy RCR/PCR files have a separate format (no `paths` attribute, for example)... uhh let me think about this some more and put together a better solution.


## Restructuring if raw_data not saved in structured file

> If I want to troubleshoot some issues with the structuring and need the `raw_data` attribute again after it has been dropped, would I just have to use `from_file` method to repopulate it [since the paths to raw_data will still be stored?]?

I had not thought about or tested this. But yes, you'd just have to throw away your old object and restructure it from scratch using `from_file` and `structure`. Maybe I can add in some code that if the raw data path is specified but the raw data is not included, it will try to load it when using `from_json_file`. Not 100% sure this will work though. The two options then would be:

A. Keep things as they are right now. More clear code. But have to save the raw data in order to restructure it.
B. Try to autoload raw_data from paths if raw_data is not present. Less clear code. Do not have to save the raw data in order to restructure it.

@patrickherring-TRI @chirranjeevigopal-TRI do you think A or B is better?

Reply:
> ## Differences in naming
> Good observation, I will change them to be the same. Would it be ok with you to just call everything `as_legacy`? Or alternatively, `omit_raw`? If `omit_raw` I'd want to change the BEEPDatapath syntax to all be `omit_raw` instead of `legacy`.
> 
> On second thought, that might make things really complicated, since the legacy RCR/PCR files have a separate format (no `paths` attribute, for example)... uhh let me think about this some more and put together a better solution.

Yeah looking through the base structure, I sensed that `omit_raw` might be a sub-criterion of `as_legacy` :-) but I was not entirely sure.
> 
> ## Restructuring if raw_data not saved in structured file
> > If I want to troubleshoot some issues with the structuring and need the `raw_data` attribute again after it has been dropped, would I just have to use `from_file` method to repopulate it [since the paths to raw_data will still be stored?]?
> 
> I had not thought about or tested this. But yes, you'd just have to throw away your old object and restructure it from scratch using `from_file` and `structure`. Maybe I can add in some code that if the raw data path is specified but the raw data is not included, it will try to load it when using `from_json_file`. Not 100% sure this will work though. The two options then would be:
> 
> A. Keep things as they are right now. More clear code. But have to save the raw data in order to restructure it.
> B. Try to autoload raw_data from paths if raw_data is not present. Less clear code. Do not have to save the raw data in order to restructure it.

> @patrickherring-TRI @chirranjeevigopal-TRI do you think A or B is better?
I dont think the `raw_data` has to be auto-loaded when using `from_json_file`.  I was just clarifying if the way to re-populate `raw_data` attribute for a `BeepDatapath` object after it has been dropped is to just run `from_file` with the requisite raw csv path? We could test it out after you merge.

Reply:
@ardunn I think A is a better option since trying to remember file paths is likely to cause headaches down the line and it makes the files stateful. A better option might be a helper function that can make the path and file name lookup automatic for our data pipeline since that is well known on our system. Something like a `beep.structure.base.raw_data_reload` function or maybe a function in the utils to supply the raw data path given the name of the protocol in the metadata of the structure file.

Reply:
> Yeah looking through the base structure, I sensed that omit_raw might be a sub-criterion of as_legacy :-) but I was not entirely sure.

as_legacy has been changed to omit_raw.


> I dont think the `raw_data` has to be auto-loaded when using `from_json_file`. I was just clarifying if the way to re-populate `raw_data` attribute for a `BeepDatapath` object after it has been dropped is to just run `from_file` with the requisite raw csv path? We could test it out after you merge.

As it is right now, yes: if you didn't save the raw data, you would reload and restructure the entire file from the original raw data file.

If you _did_ save the raw data, you can load it and restructure any of the attrs you're interested in, independently. For example, you can load a structured file and restructure the diagnostic parts independently of everything else.

Reply:
> @ardunn I think A is a better option since trying to remember file paths is likely to cause headaches down the line and it makes the files stateful. A better option might be a helper function that can make the path and file name lookup automatic for our data pipeline since that is well known on our system. Something like a `beep.structure.base.raw_data_reload` function or maybe a function in the utils to supply the raw data path given the name of the protocol in the metadata of the structure file.

Ok, I will leave that for a separate PR/issue then if people feel like it is needed!

Reply:
Thank you for the thoughtful responses and the great work, looks good to me!